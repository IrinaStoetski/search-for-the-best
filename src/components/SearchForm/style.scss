@import '../../styles/shared/style.scss';

.search {
  &-form {
    display: flex;
    align-items: flex-end;
    padding: 16px;
    width: 35vw;

    @include mobile{
       width: 100%;
       flex-direction: column;
       align-items: center;
    }

    @include tablet{
      width: 70vw;
    }
 
    .btn {
      margin-left: 8px;

      @include max-width($landscape-tablet-big){
        width: 50%;
        margin-top: 16px;
     }
      @include tablet{
        width: 70%;
     }
    }
  }
}
