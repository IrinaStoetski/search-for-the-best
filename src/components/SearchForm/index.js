// @flow
import React, { useContext } from 'react';
import { useFormik } from 'formik';
// ContextProviders
import { LayoutContext } from '../../contextProviders/LayoutContext';
// Components
import Input from '../Input';
import Button from '../Button';
// Styles
import './style.scss';

const SearchForm = () => {
  const layoutContextData = useContext(LayoutContext);
  const { isLoading, getRepositoriesList } = layoutContextData;
  const formik = useFormik({
    initialValues: {
      name: '',
    },
    onSubmit: (values) => {
      getRepositoriesList(values.name);
    },
  });
  return (
    <form
      className="search-form"
      onSubmit={formik.handleSubmit}
    >
      <Input
        title="Type a name and find your star"
        formik={formik}
        formikKey="name"
        onChange={formik.handleChange}
        value={formik.values.name}
      />
      <Button
        buttonType="submit"
        loading={isLoading}
      >
        Search
      </Button>
    </form>
  );
};

export default SearchForm;
