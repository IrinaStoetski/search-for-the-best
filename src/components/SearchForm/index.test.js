/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { LayoutContext } from '../../contextProviders/LayoutContext';
import SearchForm from './index';

test('renders correctly SearchForm component', () => {
  const getRepositoriesList = () => null;
  const isLoading = false;
  const activeQuery = {};

  const tree = renderer
    .create(
      <LayoutContext.Provider
        value={{
          getRepositoriesList,
          isLoading,
          activeQuery,
        }}
      >
        <SearchForm />
      </LayoutContext.Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
