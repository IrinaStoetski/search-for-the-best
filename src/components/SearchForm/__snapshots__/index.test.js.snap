// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly SearchForm component 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <label>
    Type a name and find your star
    <input
      className="input primary "
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <button
    className="btn primary "
    type="submit"
  >
    Search
  </button>
</form>
`;
